### ISSUE TITLE
Downloading album issue

### ISSUE THREAD

**Issue by nextgearslab** [Evolution0/bandcamp-dl#26]
nicks-mac-mini:bandcamp-dl nick$ python bandcamp-dl.py http://liluglymane.bandcamp.com/album/third-side-of-tape
Traceback (most recent call last):
  File "bandcamp-dl.py", line 56, in <module>
    album = bandcamp.parse(url)
  File "/Users/nick/Documents/bandcamp-dl-master/bandcamp-dl/Bandcamp.py", line 28, in parse
    album_meta = self.extract_album_meta_data(r)
  File "/Users/nick/Documents/bandcamp-dl-master/bandcamp-dl/Bandcamp.py", line 67, in extract_album_meta_data
    embedData = self.get_embed_string_block(request)
  File "/Users/nick/Documents/bandcamp-dl-master/bandcamp-dl/Bandcamp.py", line 100, in get_embed_string_block
    embedStringBlock = jsobj.read_js_object("var EmbedData = %s" % embedStringBlock)
  File "/Users/nick/Documents/bandcamp-dl-master/bandcamp-dl/jsobj.py", line 71, in read_js_object
    return visit(Parser().parse(code))
  File "/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/slimit/parser.py", line 56, in **init**
    self.lexer.build(optimize=lex_optimize, lextab=lextab)
  File "/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/slimit/lexer.py", line 102, in build
    self.lexer = ply.lex.lex(object=self, **kwargs)
  File "/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/ply/lex.py", line 893, in lex
    if '.' not in lextab:
TypeError: argument of type 'module' is not iterable
nicks-mac-mini:bandcamp-dl nick$

**Comment by travs** [comment]
:+1: Same here

**Comment by nextgearslab** [comment]
@iheanyi any idea why this keeps occurring?

**Comment by iraritchiemeek** [comment]
Same issue

**Comment by mystic74** [comment]
tl;dr
The following line might fix the problem
sudo pip install ply==3.4

Longer explanation:

This seems to be an issue with the ply, and the way that slimit uses it,

the issue regarding the matter can be viewed here :
https://github.com/rspivak/slimit/issues/76

the simple fix is to downgrade ply, and for me it will do for now.
The better way will be for either one of the 
two packages to solve the issue..

**Comment by nextgearslab** [comment]
@mystic74 Oh thanks a lot this fixed it! I have been using youtube-dl but I still do enjoy bandcamp-dl a lot more. Appreciate the help can't believe how long it took me to see your post damn notifications!

**Comment by ColemanGariety** [comment]
@mystic74 this works, thanks!

**Comment by cibulka** [comment]
After downgrading to `ply 3.4`, I have this error: 

```
~/Documents/repos/bandcamp-dl/bandcamp-dl/bandcamp-dl.py http://blackesteverblack.bandcamp.com/album/hide-before-dinner
unknown url type: //popplers5.bandcamp.com/download/track?enc=mp3-128&fsig=15e67e013dcc256151e75e0f8f99d993&id=368538134&stream=1&ts=1444652831.0
Downloading failed..
```

Ideas?

**Comment by rdlmda** [comment]
@cibulka I am having the same error, I believe this is a separate issue, let's discuss it on its own page (#32)

**Comment by iheanyi** [comment]
Closed in favor of #32
